src/lib/core/adaptive/screen-type.ts

```diff
@@ -0,0 +1,102 @@
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {Subject} from 'rxjs/Subject';
+
+
+export const SCREEN_TYPES = {
+  HANDSET: '(max-width: 600px) and (orientation: portrait), (max-width: 960px) and ' +
+           '(orientation: landscape)',
+  TABLET: '(min-width: 600px) and (max-width: 840px) and (orientation: portrait), ' +
+          '(min-width: 960px) and (max-width: 1280px) and (orientation: landscape)',
+  WEB: '(min-width: 840px) and (orientation: portrait), (min-width: 1280px) and ' +
+       '(orientation: landscape)',
+
+  HANDSET_PORTRAIT: '(max-width: 600px) and (orientation: portrait)',
+  TABLET_PORTRAIT: '(min-width: 600px) and (max-width: 840px) and (orientation: portrait)',
+  WEB_PORTRAIT: '(min-width: 840px) and (orientation: portrait)',
+
+  HANDSET_LANDSCAPE: '(max-width: 960px) and (orientation: landscape)',
+  TABLET_LANDSCAPE: '(min-width: 960px) and (max-width: 1280px) and (orientation: landscape)',
+  WEB_LANDSCAPE: '(min-width: 1280px) and (orientation: landscape)',
+};
+
+export interface Breakpoint {
+  observe: Observable<boolean>;
+  destroy: () => void;
+}
+
+interface InternalBreakpoint {
+  query: string;
+  subject: Subject<boolean>;
+  callback: (mql: MediaQueryList) => void;
+  usageCount: number;
+  watcher: MediaQueryList;
+}
+
+@Injectable()
+export class ScreenType {
+  private breakpoints: Map<string, InternalBreakpoint> = new Map();
+
+  /**
+   * Registers a breakpoint to be listened for.
+   */
+  registerBreakpoint(query: string): Breakpoint {
+    let breakpoint: InternalBreakpoint;
+    if (this.breakpoints.has(query)) {
+      breakpoint = this.breakpoints.get(query);
+      breakpoint.usageCount += 1;
+      window.setTimeout(() => {
+        breakpoint.subject.next(breakpoint.watcher.matches);
+      });
+    } else {
+      let watcher = window.matchMedia(query);
```

Comment:
You may want to consider a WebKit consideration:  https://github.com/angular/flex-layout/blob/master/src/lib/media-query/match-media.ts#L147